packages/@react-spectrum/card/src/BaseLayout.tsx

```diff
@@ -84,11 +84,22 @@ export class BaseLayout<T> extends Layout<Node<T>> implements KeyboardDelegate {
   }
 
   getVisibleLayoutInfos(rect) {
+    return this._getVisibleLayoutInfos(rect);
+  }
+
+  // this is used by the up and down arrows which can't know about pesisted keys
+  _getClosetLayoutInfos(rect) {
```

Comment:
Both `getKeyBelow` and `getKeyAbove` call `_findClosest` which calls `_findClosestLayoutInfo` which used to call `getVisibleLayoutInfos`. If `getVisibleLayoutInfos` adds `persistedKeys` in it, then the closest row is the current row and the user can't use the up or down arrows to navigate the grid. I had to seperate `getVisibleLayoutInfos` into a version that knows about `persistedKeys` for general usage and a version that doesn't for the arrow key nasvigation. I'll update the message to be more clear about this.

Reply:
Is it possible to update `_findClosestLayoutInfo` so it ignores a layout info that is exactly the same as the current `target`?

Reply:
I'm nervous about that because its persistedKeys not key. When or if we have more than the currently focused, will the other persisted keys also be in the current row? Ignoring the current target wouldn't work in that case while this would.

Reply:
Actually, does `persistedKeys` even play a role here? `getVisibleLayoutInfos` would gets the layout infos that **would** be visible in the provided Rect, not the layout infos that are physically visible.

Reply:
I checked my notes and what I have is Devon suggesting I investigate making `isVisible` include `persistedKeys`. That didn't quite make sense once I looked at the code, so I adjusted `getVisibleLayoutInfos` to factor in both `isVisible` and `persistedKeys`. Devon's general suggestion that ListLayout had a better understanding of what should exist in the DOM then Virtualizer. I didn't see another way for ListLayout to add LayoutInfos than `getVisibleLayoutInfos`.